AutoSizedImage.js

```diff
@@ -14,19 +14,19 @@ export default class AutoSizedImage extends PureComponent {
   constructor(props) {
     super(props);
     this.state = {
-      // set width 1 is for preventing the warning
+      // set width 0 is for preventing mipmapping in iOS
       // You must specify a width and height for the image %s
-      width: this.props.style.width || 1,
-      height: this.props.style.height || 1,
+      width: this.props.style.width || 0,
+      height: this.props.style.height || 0,
     };
   }
 
-  componentDidMount() {
+  async componentWillMount() {
     //avoid repaint if width/height is given
-    if (this.props.style.width || this.props.style.height) {
+    if (this.props.style.width && this.props.style.height) {
       return;
     }
-    Image.getSize(this.props.source.uri, (w, h) => {
+    await Image.getSize(this.props.source.uri, (w, h) => {
```

Comment:
The use of `await` here doesn't really add anything. `await` is nice syntactic sugar for deferring the rest of the `async` function until after the promise resolves but... there isn't anything else in the function. It won't alter the speed of loading the component.

Further, I'm quite nervous about putting any async request in `componentWillMount`. There's always a risk that we'll call `setState` on an unmounted component, which will result in an error. I'm also given to understand that when Fiber rolls around, `componentWillMount` could get called more than once, or at odd times... so I think we need to use `componentDidMount` for this kind of thing.